pymongo/collection.py

```diff
@@ -3091,6 +3093,10 @@ def map_reduce(self, map, reduce, out, full_response=False, session=None, **kwar
            this collection is automatically applied to this operation (if the
            output is not inline) when using MongoDB >= 3.4.
 
+        .. versionchanged:: 3.13
+           Deprecated :meth:`map_reduce`. Support for this function is deprecated in
```

Comment:
We should also say that this is removed in PyMongo 4.0:
https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#collection-map-reduce-and-collection-inline-map-reduce-are-removed

You should say here what users should do instead. An abbreviated version of what the PyMongo 4 upgrade guide says.